tests/db_functions/math/test_exp.py

```diff
@@ -0,0 +1,44 @@
+import math
+from decimal import Decimal
+
+from django.db.models import DecimalField
+from django.db.models.functions import Exp
+from django.test import TestCase
+
+from ..models import DecimalModel, FloatModel, IntegerModel
+
+
+class ExpTests(TestCase):
+
+    def test_decimal(self):
+        DecimalModel.objects.create(n1=Decimal('-12.9'), n2=Decimal('0.6'))
+        obj = DecimalModel.objects.annotate(n1_d=Exp('n1'), n2_d=Exp('n2')).first()
+        self.assertAlmostEqual(float(obj.n1_d), math.exp(obj.n1))
+        self.assertAlmostEqual(float(obj.n2_d), math.exp(obj.n2))
```

Comment:
Adding `output_field = FloatField()` to `Exp` will mean that this doesn't need casting to `float`.